robot/impl/resource_manager.go

```diff
@@ -51,8 +51,10 @@ type resourceManager struct {
 	moduleManager  modif.ModuleManager
 	opts           resourceManagerOptions
 	logger         logging.Logger
-	configLock     sync.Mutex
-	viz            resource.Visualizer
+
+	// resourceGraphLock manages access to the resource graph and nodes. If either may change, this lock should be taken.
+	resourceGraphLock sync.Mutex
```

Comment:
nice rename + clarifying comment - i always found `configLock` to be a bit confusing